packages/icon/src/Icon.tsx

```diff
@@ -59,19 +64,23 @@ export const createIcon = (children: ReactNode, name: string) => {
 		style,
 		weight = 'regular',
 	}: IconProps) => {
-		const resolvedSize = `${iconSizes[size]}rem`;
-		const resolvedWeight = iconWeights[weight][size];
+		const resolvedSize = mapResponsiveProp(size, (s) => `${iconSizes[s]}rem`);
+		const resolvedWeight = mapResponsiveProp(
```

Comment:
Is the weight meant to be responsive too? If not, we don't need to use `mapResponsiveProp` here